src/shared/condition.c

```diff
@@ -556,18 +557,26 @@ static int condition_test_needs_update(Condition *c) {
                 return true;
 
         /* If the file system is read-only we shouldn't suggest an update */
-        if (path_is_read_only_fs(c->parameter) > 0)
+        r = path_is_read_only_fs(c->parameter);
+        if (r < 0)
+                log_debug_errno(r, "Failed to determine if '%s' is read-only, ignoring: %m", c->parameter);
+        if (r > 0)
                 return false;
 
         /* Any other failure means we should allow the condition to be true, so that we rather invoke too
          * many update tools than too few. */
 
         p = strjoina(c->parameter, "/.updated");
-        if (lstat(p, &other) < 0)
+        if (lstat(p, &other) < 0) {
+                if (errno != ENOENT)
+                        log_debug_errno(errno, "Failed to stat() '%s', ignoring: %m", p);
```

Comment:
"ignoring" → "assuming an update is needed"? Not that it matters too much, since this is a very unlikely error.